mage_ai/data_cleaner/shared/utils.py

```diff
@@ -59,6 +63,30 @@ def is_spark_dataframe(df):
     return type(df).__module__ == 'pyspark.sql.dataframe'
 
 
+def __parse_element(element: str) -> Any:
+    if element == 'nan' or element == 'np.nan':
+        return np.nan
+    else:
+        try:
+            return eval(element)
+        except:
+            return None
+
+
+def parse_list(list_literal: Union[str, List[Any]]) -> FrozenList:
+    dtype = type(list_literal)
+    if dtype is FrozenList:
+        return list_literal
+    elif dtype in (list, tuple):
+        return FrozenList(list_literal)
+    elif dtype is not str:
+        return None
+    list_literal = list_literal.strip('[]() ')
+    if list_literal == '':
+        return FrozenList([])
+    return FrozenList([__parse_element(element) for element in list_literal.split(', ')])
```

Comment:
what if there's no space between elements, like '1,2,3,4'?

Reply:
Expanded the splitting algorithm to now count whitespace as well.